sdk/keyvault/Azure.Security.KeyVault.Secrets/src/SecretClient.cs

```diff
@@ -140,7 +140,79 @@ public virtual Response<KeyVaultSecret> GetSecret(string name, string version =
         }
 
         /// <summary>
-        /// Lists the properties of all enabled and disabled versions of the specified secret. You can use the returned <see cref="SecretProperties.Name"/> and <see cref="SecretProperties.Version"/> in subsequent calls to <see cref="GetSecretAsync"/>.
+        /// Get a specified secret from a given key vault, but does not throw an exception if the secret does not exist.
+        /// </summary>
+        /// <remarks>
+        /// The get operation is applicable to any secret stored in Azure Key Vault.
+        /// This operation requires the secrets/get permission.
+        /// </remarks>
+        /// <param name="name">The name of the secret.</param>
+        /// <param name="version">The version of the secret.</param>
+        /// <param name="context">A <see cref="RequestContext"/> controlling the request lifetime, error handling, and per-call pipeline policies.</param>
+        /// <returns>A response containing the <see cref="KeyVaultSecret"/> or <c>null</c> if not found.</returns>
+        /// <exception cref="ArgumentException"><paramref name="name"/> is an empty string.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="name"/> is null.</exception>
+        /// <exception cref="RequestFailedException">The server returned an error. See <see cref="Exception.Message"/> for details returned from the server.</exception>
+#pragma warning disable AZC0002 // DO ensure all service methods, both asynchronous and synchronous, take an optional CancellationToken parameter called cancellationToken.
+        public virtual async Task<NullableResponse<KeyVaultSecret>> GetSecretAsync(string name, string version, RequestContext context)
+#pragma warning restore AZC0002 // DO ensure all service methods, both asynchronous and synchronous, take an optional CancellationToken parameter called cancellationToken.
```

Comment:
Curious: is this warning suppression still needed?  I had thought we did some work to allow this type of method without an analyzer warning.

Reply:
Yep. I wasn't able to compile successfully without it. Perhaps it's because `context` isn't optional and the validation error isn't that specific? Just a guess.

Reply:
Cool, thanks for the explanation!  I'll update my mental model.